framework/SupervisedLearning/KerasRegression.py

```diff
@@ -0,0 +1,645 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on Dec. 20, 2018
+  @author: wangc
+  base class for tensorflow and keras used for deep neural network
+  i.e. Multi-layer perceptron classifier, CNN, LSTM
+"""
+#for future compatibility with Python 3--------------------------------------------------------------
+from __future__ import division, print_function, unicode_literals, absolute_import
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import copy
+import numpy as np
+import random as rn
+import matplotlib
+import platform
+from scipy import stats
+import os
+import utils.importerUtils
+tf = utils.importerUtils.importModuleLazyRenamed("tf", globals(), "tensorflow")
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from .SupervisedLearning import supervisedLearning
+#Internal Modules End--------------------------------------------------------------------------------
+
+class KerasRegression(supervisedLearning):
```

Comment:
I think this class need to be merged with the KerasClassifier class. Most of the functionalities are the same for both Regression and Classifier. I think a new base class can be extracted from these two classes.

Reply:
I completely agree (this is one of the reasons this is marked Do Not Merge).  My plan is to make a KerasBaseClass, and put the common things in KerasRegression and KerasClassifier in it.

Reply:
I have created a base class and pulled some common features out to it. I am still deciding if I need to work a bit more to get more common functions out or not.

Reply:
I think I have extracted pretty much all of the common functionality into a base class.  So if you want to take a look now, it would be welcome.